tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/demo/tf-interactive-inference-demo.html

```diff
@@ -340,131 +340,169 @@
-                  idx++
-                ) {
-                  tlist.push(this.convertExToTensor(examples[idx]));
-                }
-                const tconcat = tf.concat(tlist);
-                tlist.forEach((tensor) => tensor.dispose());
-                const input = tconcat.reshape([tlist.length, 104]);
-                const res = this.model.predict(input, {batchSize: BATCH_SIZE});
-                const vals = await res.data();
-                predValuesList.push(vals);
-                input.dispose();
-                res.dispose();
-                tconcat.dispose();
-              }
-              const predSize = predValuesList.reduce((a, b) => a + b.length, 0);
-              const predValues = new Float32Array(predSize);
-              let curIdx = 0;
-              for (let i = 0; i < predValuesList.length; i++) {
-                predValues.set(predValuesList[i], curIdx);
-                curIdx += predValuesList[i].length;
-              }
-              for (let trialIdx = 0; trialIdx < numTrials; trialIdx++) {
-                const startingPredIdx =
-                  trialIdx * exampleIndices.length * PRED_SIZE;
-                const scores = [];
-                for (
-                  let exampleIdx = 0;
-                  exampleIdx < exampleIndices.length;
-                  exampleIdx++
-                ) {
-                  scores.push(
-                    predValues[startingPredIdx + exampleIdx * PRED_SIZE]
-                  );
+              const data = await this.createPdPlotData(
+                e.detail.feature_name,
+                e.detail.example_index,
+                +e.detail.x_min,
+                +e.detail.x_max
+              );
+              let isNum = false;
+              for (let entry of this.$.dash.partialDepPlotEligibleFeatures) {
+                if (entry.name == e.detail.feature_name && !entry.samples) {
+                  isNum = true;
+                  break;
                 }
-                const score =
-                  scores.reduce((prev, cur) => prev + cur, 0) / scores.length;
-                const ex = examples[trialIdx * exampleIndices.length];
-                const step = isNum
-                  ? ex.features.feature[e.detail.feature_name].floatList
-                      .value[0]
-                  : [
-                      atob(
-                        ex.features.feature[e.detail.feature_name].bytesList
-                          .value[0]
-                      ),
-                    ];
-                results.push({step: step, scalar: score});
               }
-
               this.$.dash.makeChartForFeature(
                 isNum ? 'numeric' : 'categorical',
                 e.detail.feature_name,
-                [[{'1': results}]]
+                data
               );
             };
             setTimeout(method, 50);
           });
+          this.$.dash.addEventListener('sort-eligible-features', async (e) => {
```

Comment:
Can all this duplicated code be factored out?

Reply:
agreed. i would like to factor out all the duplicated code between demos, not just for this new feature, but didn't want to bog down this review with that whole effort.